lib/charms/hydra/v0/oauth.py

```diff
@@ -439,30 +437,12 @@ def get_client_secret(self, client_secret_id: str) -> Secret:
         client_secret = self.model.get_secret(id=client_secret_id)
         return client_secret
 
-    def get_client_credentials(self, relation_id: Optional[int] = None) -> OauthClientCredentials:
-        """Get the client credentials."""
-        try:
-            relation = self.model.get_relation(
-                relation_name=self._relation_name, relation_id=relation_id
-            )
-        except TooManyRelatedAppsError:
-            raise RuntimeError("More than one relations are defined. Please provide a relation_id")
-
-        data = _load_data(relation.data[relation.app], OAUTH_PROVIDER_JSON_SCHEMA)
-
-        client_id = data.get("client_id")
-        client_secret_id = data.get("client_secret_id")
-        if not client_id or not client_secret_id:
-            return None
-
-        _client_secret = self.get_client_secret(client_secret_id)
-        client_secret = _client_secret.get_content()[CLIENT_SECRET_FIELD]
-        return OauthClientCredentials(client_id, client_secret)
-
     def update_client_config(
         self, client_config: ClientConfig, relation_id: Optional[int] = None
     ) -> None:
         """Update the client config stored in the object."""
+        if len(self.model.relations) == 0:
+            return
```

Comment:
Isn't this checked on https://github.com/canonical/hydra-operator/pull/63/files#diff-747d8e831b4929f2bce44463876e22478664105f47ed7b3f05ca14182b26a250R385 ?

Reply:
Good point, I removed the duplicated check. Thanks